packages/instantsearch.js/src/lib/routers/history.ts

```diff
@@ -125,6 +133,7 @@ class BrowserHistory<TRouteState> implements Router<TRouteState> {
     this._start = start;
     this._dispose = dispose;
     this._push = push;
+    this._cleanUrlOnDispose = cleanUrlOnDispose;
```

Comment:
in the constructor we can warn if the value isn't set (in dev) to make people more aware of the change they should do. also all examples then should have the option set to show the right example.

Reply:
There's now an informational message in dev (similar to what we have for `preserveSharedStateOnUnmount`).

On updating examples, there's still the particular case of `routing: true` which I've left as-is. This means they will still write on dispose, and show a message on the console. I've chosen not to update them as I think explicitly adding the router with the option clashes with the simplicity of this approach, and users will still have to learn about the change in behaviour in migration notes / codemods for next major version.